src/cmd_all/src/single_node.rs

```diff
@@ -134,12 +158,29 @@ pub fn map_single_node_opts_to_standalone_opts(opts: &SingleNodeOpts) -> ParsedS
     if let Some(compactor_addr) = &opts.compactor_addr {
         compactor_opts.listen_addr = compactor_addr.clone();
     }
-    ParsedStandaloneOpts {
+
+    // Override with node-level extra options
+    fn update_extra_opts(opts: &mut impl Parser, extra_opts: &Option<String>) -> Result<()> {
+        if let Some(extra_opts) = extra_opts {
+            let extra_opts = split(extra_opts)?;
+            // This hack is required, because `update_from` treats arg[0] as the command name.
+            let prefix = "".to_string();
+            let extra_opts = iter::once(&prefix).chain(extra_opts.iter());
+            opts.update_from(extra_opts);
```

Comment:
According to my tests, this will also overwrite all the fields with default values, if present. Will this be a problem if we're going to provide some non-default value from the profile?

Reply:
Hmm what do you mean?

`single_node` provides some defaults, seems like that's still preserved:

https://github.com/risingwavelabs/risingwave/pull/15044/files#r1481228711

Reply:
https://github.com/risingwavelabs/risingwave/blob/482ded43baf20f81456eae0fd693f5cd17e1fe58/src/cmd_all/src/single_node.rs#L187

The default I'm referring to are here.

Reply:
By "default value" I'm not referring to `fn default_xx_opts`, but the fields attributed with `#[clap(default_value = "..")]`.

Would you mind trying...

```
--meta-addr 127.0.0.1:8888
--meta-extra-opts=--advertise-addr 127.0.0.1:9999
```

to see if `8888` gets preserved?

Reply:
Oh in this case I expect it to be overridden.

Reply:
Because the precedence is like:
`profile defaults < process options < node level extra opts`.

For `--meta-addr` it's a process level option.
So `--meta-extra-opts` should override it for meta node.

Otherwise user should not have supplied it.

Reply:
I'm not sure whether I have made clear, but here I intentionally overwrite the **advertise address** instead of the **listen address**. They are different options.

To make it more clear, you can try passing an empty string as the extra options. The `8888` could still be overwritten. Is this also expected?

Reply:
Added it: https://github.com/risingwavelabs/risingwave/pull/15044/commits/62c2310063c4e7d66a1857917ecf7e068a706ed3

Reply:
Oh yeah you're right... 8888 is not preserved in the `listen-addr`:
```
                            listen_addr: "127.0.0.1:5690",
```

Reply:
I guess this PR is blocked here now.

Reply:
Yes. I don't have any other ideas for now. Probably have to make a PR to clap to try and make a case for this.

Otherwise, I think we have to generate a node options builder per node, and use that to update the base options per node.